From 0c2aebab2d1c3cf7a02c16dd6a5af4ae55a69995 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 28 Apr 2023 12:28:19 +0100
Subject: [PATCH] RHEL: Remove --block-driver option

Go back to the old default of always installing virtio-blk drivers in
Windows guests.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2190387
---
 docs/virt-v2v-in-place.pod | 10 ----------
 docs/virt-v2v.pod          | 10 ----------
 in-place/in_place.ml       | 11 +----------
 tests/Makefile.am          |  1 -
 v2v/v2v.ml                 | 11 +----------
 5 files changed, 2 insertions(+), 41 deletions(-)

diff --git a/docs/virt-v2v-in-place.pod b/docs/virt-v2v-in-place.pod
index 6c02a99c..3d0d1b28 100644
--- a/docs/virt-v2v-in-place.pod
+++ b/docs/virt-v2v-in-place.pod
@@ -72,16 +72,6 @@ Display help.
 
 See I<--network> below.
 
-=item B<--block-driver> B<virtio-blk>
-
-=item B<--block-driver> B<virtio-scsi>
-
-When choosing a block driver for Windows guests, prefer C<virtio-blk> or
-C<virtio-scsi>.  The default is C<virtio-blk>.
-
-Note this has no effect for Linux guests at the moment.  That may be
-added in future.
-
 =item B<--colors>
 
 =item B<--colours>
diff --git a/docs/virt-v2v.pod b/docs/virt-v2v.pod
index fde60a8c..0e68de10 100644
--- a/docs/virt-v2v.pod
+++ b/docs/virt-v2v.pod
@@ -212,16 +212,6 @@ The options are silently ignored for other input methods.
 
 See I<--network> below.
 
-=item B<--block-driver> B<virtio-blk>
-
-=item B<--block-driver> B<virtio-scsi>
-
-When choosing a block driver for Windows guests, prefer C<virtio-blk> or
-C<virtio-scsi>.  The default is C<virtio-blk>.
-
-Note this has no effect for Linux guests at the moment.  That may be
-added in future.
-
 =item B<--colors>
 
 =item B<--colours>
diff --git a/in-place/in_place.ml b/in-place/in_place.ml
index 604a662d..a91ee39d 100644
--- a/in-place/in_place.ml
+++ b/in-place/in_place.ml
@@ -49,7 +49,6 @@ let rec main () =
 
   let bandwidth = ref None in
   let bandwidth_file = ref None in
-  let block_driver = ref None in
   let input_conn = ref None in
   let input_format = ref None in
   let input_password = ref None in
@@ -162,8 +161,6 @@ let rec main () =
   let argspec = [
     [ S 'b'; L"bridge" ], Getopt.String ("in:out", add_bridge),
                                     s_"Map bridge ‘in’ to ‘out’";
-    [ L"block-driver" ], Getopt.String ("driver", set_string_option_once "--block-driver" block_driver),
-                                    s_"Prefer 'virtio-blk' or 'virtio-scsi'";
     [ S 'i' ],       Getopt.String (input_modes, set_input_mode),
                                     s_"Set input mode (default: libvirt)";
     [ M"ic" ],       Getopt.String ("uri", set_string_option_once "-ic" input_conn),
@@ -230,12 +227,6 @@ read the man page virt-v2v-in-place(1).
 
   (* Dereference the arguments. *)
   let args = List.rev !args in
-  let block_driver =
-    match !block_driver with
-    | None | Some "virtio-blk" -> Virtio_blk
-    | Some "virtio-scsi" -> Virtio_SCSI
-    | Some driver ->
-       error (f_"unknown block driver ‘--block-driver %s’") driver in
   let customize_ops = get_customize_ops () in
   let input_conn = !input_conn in
   let input_mode = !input_mode in
@@ -298,7 +289,7 @@ read the man page virt-v2v-in-place(1).
 
   (* Get the conversion options. *)
   let conv_options = {
-    Convert.block_driver = block_driver;
+    Convert.block_driver = Virtio_blk;
     keep_serial_console = true;
     ks = opthandle.ks;
     network_map;
diff --git a/tests/Makefile.am b/tests/Makefile.am
index f9c748ed..c0aa8498 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -57,7 +57,6 @@ TESTS_ENVIRONMENT = $(top_builddir)/run --test
 
 TESTS = \
 	test-bad-networks-and-bridges.sh \
-	test-block-driver.sh \
 	test-cdrom.sh \
 	test-checksum-bad.sh \
 	test-checksum-good-qcow2.sh \
diff --git a/v2v/v2v.ml b/v2v/v2v.ml
index 5d3a9dbb..2e8071db 100644
--- a/v2v/v2v.ml
+++ b/v2v/v2v.ml
@@ -48,7 +48,6 @@ let rec main () =
 
   let bandwidth = ref None in
   let bandwidth_file = ref None in
-  let block_driver = ref None in
   let input_conn = ref None in
   let input_format = ref None in
   let input_password = ref None in
@@ -201,8 +200,6 @@ let rec main () =
                                     s_"Set bandwidth dynamically from file";
     [ S 'b'; L"bridge" ], Getopt.String ("in:out", add_bridge),
       s_"Map bridge ‘in’ to ‘out’";
-    [ L"block-driver" ], Getopt.String ("driver", set_string_option_once "--block-driver" block_driver),
-                                    s_"Prefer 'virtio-blk' or 'virtio-scsi'";
     [ S 'i' ],       Getopt.String (input_modes, set_input_mode),
       s_"Set input mode (default: libvirt)";
     [ M"ic" ],       Getopt.String ("uri", set_string_option_once "-ic" input_conn),
@@ -295,12 +292,6 @@ read the man page virt-v2v(1).
 
   (* Dereference the arguments. *)
   let args = List.rev !args in
-  let block_driver =
-    match !block_driver with
-    | None | Some "virtio-blk" -> Virtio_blk
-    | Some "virtio-scsi" -> Virtio_SCSI
-    | Some driver ->
-       error (f_"unknown block driver ‘--block-driver %s’") driver in
   let customize_ops = get_customize_ops () in
   let input_conn = !input_conn in
   let input_mode = !input_mode in
@@ -413,7 +404,7 @@ read the man page virt-v2v(1).
 
   (* Get the conversion options. *)
   let conv_options = {
-    Convert.block_driver = block_driver;
+    Convert.block_driver = Virtio_blk;
     keep_serial_console = not remove_serial_console;
     ks = opthandle.ks;
     network_map;
